src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java

```diff
@@ -535,4 +543,78 @@ public HtmlId forHeading(CharSequence headingText, Set<String> headingIds) {
         }
         return HtmlId.of(idValue);
     }
+
+    private final Map<ExecutableElement, HtmlId> ids = new HashMap<>();
+
+    /*
+     * Returns an id to a constructor or a method, from a centralised
+     * registry. Use to get an anchor to a constructor or method.
+     *
+     * The goal is to provide coordination, not cache.
+     */
+    private HtmlId forExecutable(ExecutableElement e) {
+        HtmlId htmlId = ids.get(e);
+        if (htmlId != null)
+            return htmlId;
+        if (e.getKind() != ElementKind.CONSTRUCTOR
+                && e.getKind() != ElementKind.METHOD)
+            throw new IllegalArgumentException(String.valueOf(e.getKind()));
+        var vmt = configuration.getVisibleMemberTable((TypeElement) e.getEnclosingElement());
```

Comment:
Will the vmt differ if we specify `-private`? And say if we have 2 methods like:
```java
private <T extends Runnable> void method(T arg) {}
public <T extends Number> void method(T arg) {}
```
Will the Number public method be consistently erased whether or not `-private` is set?

Reply:
The answer to your latter question is no. The id that will be assigned to that public method do depend on whether the private overload positioned before it is documented. You cannot do much about it with the current design of `VisibleMemberTable` without unduly complicating this PR. On the bright side, I reckon such a case is extremely rare.

Reply:
That's actually a sign of relief that the same elements will have the same id with the same input source code

Reply:
Reading your reply, I think that one of us is likely confused. Your question to which I replied "no" was this:

> Will the Number public method be consistently erased whether or not -private is set?

I replied "no", which means that the id that the public method has DO depend on whether `-private` is specified. Let me expand on that.

If `-private` is specified, the "Method Details" section will have this for the public method:

    <section class="detail" id="method(java.lang.Number)">
    <h3>method</h3>
    
and this for the private method:

    <section class="detail" id="method(T)">
    <h3 id="method(java.lang.Runnable)">method</h3>

Otherwise, the "Method Details" section will only have this for the public method:

    <section class="detail" id="method(T)">
    <h3 id="method(java.lang.Number)">method</h3>

As you can see, depending on whether `-private` is specified, the section id for the public method will be different. 
    
Is that how you understood my reply and felt relieved after that?

Reply:
It's fine, we always have an anchor to the erased version.